Source/Dafny/Translator.cs

```diff
@@ -8342,14 +8433,16 @@ AND  Apply(f,h0,bxs) == Apply(f,h0,bxs)
           var f = BplBoundVar("f", predef.HandleType, bvars);
           var boxes = Map(Enumerable.Range(0, arity), i => BplBoundVar("bx" + i, predef.BoxType, bvars));
 
-          var isness = BplAnd(
-            Snoc(Map(Enumerable.Range(0, arity), i =>
-              BplAnd(MkIs(boxes[i], types[i], true),
-                CommonHeapUse && !FrugalHeapUse ? MkIsAlloc(boxes[i], types[i], h0, true) : Bpl.Expr.True)),
-            BplAnd(MkIs(f, ClassTyCon(ad, types)),
-              CommonHeapUse && !FrugalHeapUse ? MkIsAlloc(f, ClassTyCon(ad, types), h0) : Bpl.Expr.True)));
+          var isAlloc = MkIsAlloc(f, ClassTyCon(ad, types), h0);
+          var isness = commonNotFrugalHeapUse
+            ? BplAnd(
+              Snoc(Map(Enumerable.Range(0, arity), i =>
+                  BplAnd(MkIs(boxes[i], types[i], true), MkIsAlloc(boxes[i], types[i], h0, true))),
+                BplAnd(MkIs(f, ClassTyCon(ad, types)), isAlloc)))
+            : MkArityEq(f, arity);
 
-          Action<Bpl.Expr, string> AddFrameForFunction = (hN, fname) => {
+          Action<Bpl.Expr, ReqReadsApp, int> AddFrameForFunction = (hN, rra, adArity) =>
+          {
```

Comment:
Move this curly brace to the end of the previous line.
Ditto for lines 8458 and 8470.